app/helpers/auth.rb

```diff
@@ -0,0 +1,39 @@
+ helpers do
+
+  def current_user
+    if session[:user_id]
+      @current_user ||= User.find_by_id(session[:user_id])
```

Comment:
Even better, only reach into the session if `@current_user` has not already been assigned:
```ruby
def current_user
  @current_user ||= User.find_by_id(session[:user_id])
end
```